### ISSUE TITLE
CSV Imports Wrong Status

### ISSUE THREAD

**Issue by lbunge** [snipe/snipe-it#3531]
#### Expected Behavior (or desired behavior if a feature request)

Upload a CSV via the interface import. If status in csv is left blank but has a username, then it is assigned to that user with status "Deployed"

-----

#### Actual Behavior

Upload a CSV via the interface import. Status is blank, username is present, item gets assigned to the user, but the status label is Pending. On the dashboard, it says I have all the item deployed in the chart, but they only appear in Pending under the assets status
-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running
3.6.4 build 12
- What OS and web server you're running Snipe-IT on
Mac OSX 10.11, ubuntu 16.04
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
install.sh
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
Importer and Asset Statuses
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
No errors
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
CSV not allowed in demo
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database
No
![screen shot 2017-05-03 at 1 05 28 pm](https://cloud.githubusercontent.com/assets/28149061/25672372/3b79b9d6-3001-11e7-8d01-3a5a3b87c163.png)
![screen shot 2017-05-03 at 1 05 15 pm](https://cloud.githubusercontent.com/assets/28149061/25672373/3b7c18f2-3001-11e7-9853-9c7da9029b2d.png)

**Comment by snipe** [comment]
I'm confused - in the screenshot you provided, it's not set to Pending, but rather to Deployed.

**Comment by lbunge** [comment]
One of the screenshots is the display of all assets, which it does display Deployed. But the first screenshot is of the asset details page of that same asset with the status saying Pending. It's basically set to two different statuses.

**Comment by snipe** [comment]
Ah, I understand now, thanks.

That... doesn't really make any sense though, since there's only one status_label field that the data pulls from. We don't have a secondary status ID column in any of the tables that could/would override anything like that.

What's the status_id map to in your assets table in the database?

**Comment by nicqrocks** [comment]
Hi, I work with @lbunge. According to the database the asset is assigned to a user, however it has the `status_id` of 1. This is the query I ran to figure this out.
```
MariaDB [snipeit]> select a.asset_tag, a.assigned_to, u.username, a.status_id, sl.name, sl.deployable, sl.pending
from assets a join status_labels sl on a.status_id = sl.id join users u on a.assigned_to = u.id limit 10;
+-----------+-------------+-------------+-----------+---------+------------+---------+
| asset_tag | assigned_to | username    | status_id | name    | deployable | pending |
+-----------+-------------+-------------+-----------+---------+------------+---------+
| 00543     |         559 | techsupport |         1 | Pending |          0 |       1 |
| 00536     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00541     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00538     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00540     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00606     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00549     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00558     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00539     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
| 00551     |         667 | 5ahs        |         1 | Pending |          0 |       1 |
+-----------+-------------+-------------+-----------+---------+------------+---------+```

**Comment by cornwella** [comment]
Hello, here's another one of @lbunge's co-workers. The only files we've modified from a stock install is the `snipeit/app/Http/Controllers/ReportsController.php` file to allow us to export the internally generated auto-incrementing asset IDs in our CSV reports.

The modifications we made to this file were:
`
$headers=[
    trans('general.company'),
    trans('admin/hardware/table.asset_tag'),
    'ID',
    trans('admin/hardware/form.manufacturer'),
`
and
`
foreach ($assets as $asset) {
    // Add a new row with data
    $values=[
        ($asset->company) ? $asset->company->name : '',
        $asset->asset_tag,
        ($asset->id) ? $asset->id : '',
        ($asset->model->manufacturer) ? $asset->model->manufacturer->name : '',
`
... which should only affect CSV exports. The rest is a stock install (beyond custom fields generated in the UI).

**Comment by cornwella** [comment]
This can be marked as closed. We had a single custom field data type in our database that was `ANY` instead of being blank, causing this whole mess. It works as intended now.

**Comment by lbunge** [comment]
Maybe we didn't quite understand the import csv instructions found here (https://snipe-it.readme.io/v3.6.2/docs/importing). We uploaded each asset with a blank status and a filled in username assuming that it would mark it as the "Deployed" status label and check it out to the user. It would always checkout the asset to the user, but mark the asset with the "Pending" status label presumably because the status_id was 1. Since "Pending" isn't deployable it caused us some problems.

So we created a status label called In Use and fill all the assets status label in the csv with it. It still marks it as deployed and checked out to the user but it also gives it the label of In Use.
